### PR TITLE
Extract CheckpointAcceptnaceService and migrate CheckpointBlockValidator

### PR DESCRIPTION
**Pull request by wkoszycki**
to F[_]

### CODE REVIEW

File: src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    val parents = getParents(cb)
+
+    val validate: F[Ior[NonEmptyList[CheckpointBlockValidation], AddressBalance]] = for {
+      x <- parents.traverse(z => validateCheckpointBlockTree(z))
+      folded = x
+        .foldLeft(Map.empty[String, Long].rightIor[NonEmptyList[CheckpointBlockValidation]])(
+          (result, d) => result.combine(d)
+        )
+        .map(getSummaryBalance(cb) |+| _)
+      result = folded.flatMap(diffs => diffs.rightIor) // TODO: wkoszycki   how to make diffs.forallM(validateDiff)
```
Comment (wkoszycki) [inline src/main/scala/org/constellation/checkpoint/CheckpointBlockValidator.scala]: @marcinwadon :sos:

File: src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -23,7 +23,7 @@
         <appender-ref ref="console" />
     </logger>
 
-    <root level="info">
+    <root level="debug">
```
Comment (mchrapek) [inline src/test/resources/logback-test.xml]: intentional change?

File: src/test/scala/org/constellation/checkpoint/CheckpointBlockValidatorNelTest.scala [hunk 3]
```diff
@@ -68,19 +76,18 @@ class CheckpointBlockValidatorNelTest
...
     rightParent.transactions shouldReturn Seq.empty
 
-    checkpointService.fullData(rightParent.baseHash) shouldReturn IO.pure(Some(CheckpointCache(Some(rightParent))))
-    checkpointService.fullData(leftParent.baseHash) shouldReturn IO.pure(Some(CheckpointCache(Some(leftParent))))
-    checkpointService.memPool shouldReturn mock[CheckpointBlocksMemPool[IO]]
-    checkpointService.memPool.size() shouldReturn IO.pure(0)
+//    checkpointService.fullData(rightParent.baseHash) shouldReturn IO.pure(Some(CheckpointCache(Some(rightParent))))
+//    checkpointService.fullData(leftParent.baseHash) shouldReturn IO.pure(Some(CheckpointCache(Some(leftParent))))
+//    checkpointService.memPool shouldReturn mock[CheckpointBlocksMemPool[IO]]
+//    checkpointService.memPool.size() shouldReturn IO.pure(0)
```
Comment (mchrapek) [inline src/test/scala/org/constellation/checkpoint/CheckpointBlockValidatorNelTest.scala]: Can it be removed?

File: src/test/scala/org/constellation/checkpoint/CheckpointBlockValidatorNelTest.scala [hunk 4]
```diff
@@ -89,7 +96,7 @@ class CheckpointBlockValidatorNelTest
     rightBlock.transactions shouldReturn Seq(tx3, tx4)
 
     dao.snapshotService shouldReturn snapService
-    dao.checkpointService shouldReturn checkpointService
+//    dao.checkpointService shouldReturn checkpointService
```
Comment (mchrapek) [inline src/test/scala/org/constellation/checkpoint/CheckpointBlockValidatorNelTest.scala]: Can it be removed?

File: src/test/scala/org/constellation/checkpoint/CheckpointBlockValidatorNelTest.scala [hunk 5]
```diff
@@ -129,23 +136,24 @@ class CheckpointBlockValidatorNelTest
...
 
   test("it should get transactions from parent") {
     rightParent.transactions shouldReturn Seq(tx2)
-    val ancestors = Seq("ancestor_in_snap")
-    rightParent.parentSOEBaseHashes() shouldReturn ancestors
-    checkpointService.fullData("ancestor_in_snap") shouldReturn IO.pure(None)
+    val ancestors = List("ancestor_in_snap")
+    checkpointParentService.getParents(rightParent) shouldReturnF List()
+    checkpointParentService.getParents(rightBlock) shouldReturnF List(rightParent)
+//    checkpointService.fullData("ancestor_in_snap") shouldReturn IO.pure(None)
```
Comment (mchrapek) [inline src/test/scala/org/constellation/checkpoint/CheckpointBlockValidatorNelTest.scala]: Can it be removed?

**Pull request merged**